### DISCUSSION TITLE
Confusion about HeaderFilterRegex in .clang-tidy

### DISCUSSION THREAD

**Discussion by xuke-hat** [duckdb/duckdb#9537] [General]
In .clang-tidy:

```
HeaderFilterRegex: '.*^(re2.h)'
```

This only will match files that named `re2.h`, is this intended?

**Comment by Tishj** [comment]
any paths that don't end in `re2.h` no?
But I'm not sure of the origins of that file

**Reply by xuke-hat** [reply]
I guess the writer wants to match all files except those that end with `re2.h`. But here `^` matches a line start, so to my understanding this will only match `re2.h`, is it?

I found a [related stackoverflow thread](https://stackoverflow.com/questions/71797349/is-it-possible-to-ignore-a-header-with-clang-tidy).

**Comment by xuke-hat** [comment]
@Mytherin Would you mind check this?

**Comment by Tishj** [comment]
Why is this relevant?

**Reply by xuke-hat** [reply]
This results in many header files not checked by clang-tidy.

**Reply by Tishj** [reply]
Hmm I see
We've had plenty of clang tidy issues flagged in headers though, so that's confusing

**Reply by Tishj** [reply]
But I do think you might be on to something
Happy to review a PR that removes this setting and instead explicitly disables the linter for the re2.h file, I think this can be done with comments to turn off and on

Maybe this is a fruitful possibility
https://stackoverflow.com/a/56182569

Or a variation of this:
https://stackoverflow.com/a/75586089

**Reply by xuke-hat** [reply]
I runned clang-tidy after fixing the regex, but now there are a lot of errors. Would it be better to ignore all files in third_party? I will make a draft PR after fixing the easiest ones.

**Reply by Tishj** [reply]
I don't think you want to fix the errors, the draft PR will just be to highlight the problem.
I would like to see the re2.h file excluded from the clang-tidy checks, because that is what likely sparked this issue originally
